pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -91,7 +91,7 @@ func (c *BlobDiskController) CreateVolume(name, storageAccount, storageAccountTy
 		accounts = append(accounts, accountWithLocation{Name: storageAccount})
 	} else {
 		// find a storage account
-		accounts, err = c.common.cloud.getStorageAccounts(storageAccountType)
+		accounts, err = c.common.cloud.getStorageAccounts(storageAccountType, location)
```

Comment:
We can also update upcoming account matching logic when handling:
'// TODO: create a storage account and container'

Reply:
yes, I am going to fix it in next PR, it's in my plan.